Dockerfile.prow

```diff
@@ -1,6 +1,6 @@
 # Copyright Contributors to the Open Cluster Management project
 
-FROM openshift/origin-release:golang-1.13 AS builder
+FROM openshift/origin-release:golang-1.15 AS builder
```

Comment:
I found other observability repos are using `registry.ci.openshift.org/open-cluster-management/builder:go1.15-linux` as builder image, is there any special reason we use this for metric collector repo?

Reply:
@morvencao 
we inherit to use this image from telmetry client. Do you think we need to change to use the one in registry.ci.openshift-org?

Reply:
I found we have add builder image `registry.ci.openshift.org/open-cluster-management/builder:go1.15-linux-amd64` in prow config for metric-collector
https://github.com/openshift/release/blob/459db96c7779c2eb81713eadb1f732686c6102d8/ci-operator/config/open-cluster-management/metrics-collector/open-cluster-management-metrics-collector-main.yaml#L6-L10

And I'm updating it to multiarch image in https://github.com/openshift/release/pull/17696

Reply:
we may use `open-cluster-management` image as suggested by cicd.

Reply:
@morvencao @clyang82 
I updated the builder image